src/main/java/org/jboss/tm/listener/event/TransactionLifecycleEvent.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2015, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.jboss.tm.listener.event;
+
+import javax.transaction.Transaction;
+
+/**
+ * Details of a change in the disposition of an transaction that is passed to registered
+ * listeners ({@link TransactionLifecycleListener#onTransactionEvent(TransactionLifecycleEvent)}
+ *
+ * @param <E> the type of event that the transaction relates to
+ */
+public class TransactionLifecycleEvent<E extends Enum<E>> {
+    private Transaction transaction;
+    private TransactionLifecycleEventType<E> reason;
+
+    public TransactionLifecycleEvent(Transaction transaction, TransactionLifecycleEventType<E> reason) {
+        this.transaction = transaction;
+        this.reason = reason;
+    }
+
+    /**
+     *
+     * @return the transaction that the event relates to
+     */
+    public Transaction getTransaction() {
+        return transaction;
+    }
+
+    /**
+     * The reason that this event was created
+     *
+     * @return an indication of what kind of change has occurred
+     */
+    public TransactionLifecycleEventType<E> getReason() {
```

Comment:
An event has been generated to indicate that a transactions state has/is changing. This method gives the reason that the event was generated. I just feel that getReason has more explanatory power than getType.